hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/cli/GenericCli.java

```diff
@@ -123,4 +132,25 @@ protected PrintWriter out() {
   protected PrintWriter err() {
     return cmd.getErr();
   }
+
+  private String handleFileSystemException(FileSystemException e) {
+    // If reason is set, return the exception's message as it is.
+    if (e.getReason() != null) {
+      return e.getMessage();
+    }
+
+    // Otherwise, construct a custom message based on the type of exception
+    String errorMessage;
+    if (e instanceof NoSuchFileException) {
+      errorMessage = String.format("Error: File not found: %s", e.getFile());
+    } else if (e instanceof AccessDeniedException) {
+      errorMessage = String.format("Error: Access denied to file: %s", e.getFile());
+    } else if (e instanceof FileAlreadyExistsException) {
+      errorMessage = String.format("Error: File already exists: %s", e.getFile());
```

Comment:
Please use `getMessage()` instead of `getFile()`, to include `otherFile` in the message.  This is useful if the error affects that other file, e.g. when trying to copy without "overwrite" flag.

Since the custom message is always prepended, use `errorMessage = "..." + e.getMessage()` instead of `String.format()`.